### PR TITLE
kafkacat: rename to kcat and update to 1.7.1

### PR DESCRIPTION
**Pull request by slashapp**
#### Description

kafkacat has been renamed to kcat to adhere to the Apache Software Foundation's (ASF) trademark policies.

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 14.2.1 23C71 arm64
Xcode 15.1 15C65

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vs install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: net/kafkacat/Portfile [hunk 1]
```diff
@@ -11,12 +12,13 @@ maintainers         {@alexeyt820 gmail.com:alexey.trenikhin+macports} openmainta
...
+replaced_by         kcat
+revision            1
+livecheck.type      none
 
-configure.args      --enable-json
-
-depends_lib         port:yajl \
-                    port:librdkafka
+pre-configure {
+    ui_error "Please do not install this port since it has been replaced by 'kcat'."
```
Comment (pmetzger) [inline net/kafkacat/Portfile:20]: There's nothing to install. You should instead simply inform the user that the port has been replaced by kcat. But I believe the obsolete portgroup already does something like that.

File: net/kcat/Portfile [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+
+github.setup        edenhill kcat 1.7.1
+categories          net
+platforms           darwin
+license             BSD
+maintainers         {gmail.com:slashapp @slashapp} openmaintainer
```
Comment (pmetzger) [inline net/kcat/Portfile:10]: You are de facto changing the maintainer on an existing port here. Does @alexeyt820 feel okay with that?

File: net/kafkacat/Portfile (hunk 1 above)
Comment (reneeotten) [reply net/kafkacat/Portfile:20]: Indeed, see the guide how to use the "obsolete" PortGroup to replace a port. Additionally, please add a comment when the obsokete port can be removed (i.e., 1 year from now).

Comment (slashapp) [reply net/kafkacat/Portfile]: Thanks for the guidelines, I've made the changes.

File: net/kcat/Portfile (hunk 2 above)
Comment (pmetzger) [reply net/kcat/Portfile]: You aren't "adding" a maintainer. You're changing the maintainer of an old port. Effectively this is just a renaming, but you're substituting yourself for @alexeyt820, the maintainer up until now. Is that person okay with this?

**Comment by pmetzger** [comment]
You need to rebase your commit here, not do a merge. Also, you should change the commit message to "kafkacat: rename to kcat and update to 1.7.1" or something similar.

File: net/kcat/Portfile (hunk 2 above)
Comment (slashapp) [reply net/kcat/Portfile]: This port hasn't been updated in 4 years, I just wanted to update the version, nothing else.

**Comment by slashapp** [comment]
> You need to rebase your commit here, not do a merge. Also, you should change the commit message to "kafkacat: rename to kcat and update to 1.7.1" or something similar.

Done

File: net/kcat/Portfile (hunk 2 above)
Comment (pmetzger) [reply net/kcat/Portfile]: Then you didn't need to add yourself as the maintainer. In any case, if the timeout expires we'll assume @alexeyt820 is okay with it.

Comment (alexeyt820) [reply net/kcat/Portfile]: I'm ok with it

**Review by alexeyt820 — APPROVED** [review]

**Pull request merged**